### PR TITLE
[TRUS-1034] - New pattern for Client Reference Number

### PR DESCRIPTION
**Pull request by jassalrichy**
refactor input-text component for accessibility, update regex validation, apply updated content

### CODE REVIEW

File: app/assets/stylesheets/trustsfrontend-app.scss [hunk 1]
```diff
@@ -169,6 +169,7 @@
 }
 
 .agentClientRef, .declarationReferenceNumber {font-weight:700;}
+.form-control.agent-internal-reference {width: 100% !important;}
```
Comment (adamconder) [inline app/assets/stylesheets/trustsfrontend-app.scss]: Is there an alternative that we can do here? Do we need to absolutely use !important. Always shouts that our selectivity isn't quite correct.

Comment (adamconder) [reply app/assets/stylesheets/trustsfrontend-app.scss]: Also, curious why this needs to be 100%?

Comment (jassalrichy) [reply app/assets/stylesheets/trustsfrontend-app.scss]: Removed !important (was an oversight  and not required).
There is a requirement to show all 56 characters for a client ref. By default input["text"] is set to a width of 50% within its parent container (min-width 641px), hence setting this to 100% will allow the user to see the complete max input?

Comment (adamconder) [reply app/assets/stylesheets/trustsfrontend-app.scss]: Is there not an alternative css class we can set to expand the width or is this the largest available?

**Pull request merged**